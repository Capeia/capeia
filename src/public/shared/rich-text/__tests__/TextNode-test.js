// @flow
/* eslint-env jest, jasmine */

import React from 'react'
import renderer from 'react-test-renderer'

describe('<TextNode>', () => {
  it('renders correctly', () => {
    jest.deepUnmock('../TextNode')
    const Component = require('../TextNode').default
    const instance = renderer.create(
      <Component text='Hello World' styles={[]} />
    )
    let tree = instance.toJSON()
    expect(tree).toMatchSnapshot()

    const testCases = [
      {text: 'hello world', styles: ['BOLD']},
      {text: 'foo bar', styles: ['ITALIC', 'SUB']},
      {text: 'baz', styles: ['UNDERLINE', 'SUPER']}
    ]

    testCases.forEach(t => {
      instance.update(<Component text={t.text} styles={t.styles} />)
      tree = instance.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
