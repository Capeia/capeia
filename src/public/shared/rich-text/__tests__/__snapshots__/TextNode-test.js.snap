// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextNode> renders correctly 1`] = `
<span
  style={Object {}}
>
  Hello World
</span>
`;

exports[`<TextNode> renders correctly 2`] = `
<span
  style={
    Object {
      "fontWeight": "bold",
    }
  }
>
  hello world
</span>
`;

exports[`<TextNode> renders correctly 3`] = `
<span
  style={
    Object {
      "fontSize": "0.7em",
      "fontStyle": "italic",
      "verticalAlign": "sub",
    }
  }
>
  foo bar
</span>
`;

exports[`<TextNode> renders correctly 4`] = `
<span
  style={
    Object {
      "fontSize": "0.7em",
      "textDecoration": "underline",
      "verticalAlign": "super",
    }
  }
>
  baz
</span>
`;
